cni/cmd/tc-redirect-tap/main.go

```diff
@@ -376,3 +390,18 @@ type NoPreviousResultError struct{}
 func (e NoPreviousResultError) Error() string {
 	return "no previous result was found, was this plugin chained with a previous one?"
 }
+
+// extractArgs returns desired parameter value from cli args string
+// args string - cli args string("key1=val1;key2=val2)
+// argName string - desired args key name(key1)
+func extractArgs(args, argName string) string {
+	argumentsPairs := strings.Split(args, ";")
+	for _, pairStr := range argumentsPairs {
+		pair := strings.Split(pairStr, "=")
```

Comment:
Is the simple string splitting we're doing here sufficiently robust against embedded delimiter characters? In other words, if we're looking for "key1=val1;key2=val2", what happens if val1 contains a semicolon or an equal sign?

I'm perfectly happy to be told "This is not possible" or something to that effect. Just want to make sure that something is actually enforcing that, though, if it is the case.

Reply:
Actually it's a good question, do you have any idea how we can be sure that THIS will not happen?

Reply:
This string is provided to us by the CNI caller as defined in [the spec](CNI_ARGS) (search for `CNI_ARGS`).

`;` is defined as the delimiter there, so I guess the spec is also implicitly saying that `;` would be invalid in an actual key/value. So I don't think we can do much more in our code if the caller provides a key/val that contains `;`, it would just be user error but not detectable-in-general by us. The best we can do is validate that the arguments we are looking for are in the format expected.

It might be good to mention this in the comment for this function and link to the CNI spec.